main.rb

```diff
@@ -0,0 +1,64 @@
+require 'open3'
+
+def get_env_variable(key)
+    return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+ 
+def env_has_key(key)
+    value = get_env_variable(key)
+    return value unless value.nil? || value.empty?
+
+    abort("Input #{key} is missing.")
+end
+
+def run_command(command)
+  puts "@@[command] Executing: #{command}"
+  stdout_str, stderr_str, status = Open3.capture3(command)
+
+  if status.success?
+      return stderr_str.empty? ? stdout_str : stderr_str
+  else
+      puts "@@[error] Command failed:\n#{stderr_str}"
+      raise stderr_str
+  end
+end
+
+def abort_with0(message)
+  puts "@@[error] #{message}"
+  exit 0
+end
+
+def abort_with1(message)
+    puts "@@[error] #{message}"
+    exit 1
+end
+
+def get_available_java_versions()
+    available_java_versions = [8, 11, 17, 21]
+    available_java_versions.each do |java_version|
+        available_java_versions.delete(java_version) unless get_env_variable("JAVA_HOME_#{java_version}_X64")
+    end
+    available_java_versions.join(', ')
+end
+
+ac_selected_java_version = env_has_key('AC_SELECTED_JAVA_VERSION')
+puts "Selected Java version: #{ac_selected_java_version}"
+
+selected_java_version = get_env_variable("JAVA_HOME_#{ac_selected_java_version}_X64")
+if !selected_java_version
+  abort_with1("Java version #{selected_java_version} is not available on the runner. Available Java versions: #{get_available_java_versions()}.")
+end
+
+current_java_version = run_command('javac -version').match(/javac (\d+)\.\d+\.\d+/)[1]
+puts "Current Java Version: #{current_java_version}"
+
+if ac_selected_java_version == current_java_version
+    abort_with0("Current Java version is already the same as the selected Java version, #{selected_java_version}.")
```

Comment:
```suggestion
    abort_with0("Current Java version #{current_java_version} is already the same as the selected Java version #{selected_java_version}.")
```

After all warning and info messages printed, many user does not read the whole log, we need to specify the versions in the every messages

Reply:
I didn't write the two values ​ to this warning message because they are equal 😄

If we need to compare the two outputs:

Current:
![Screenshot 2024-10-25 at 12 57 52](https://github.com/user-attachments/assets/608fb41c-3b1a-4d46-916c-4a222b3ab157)


Suggestion:
![Screenshot 2024-10-25 at 12 57 13](https://github.com/user-attachments/assets/38b02cf6-c6b6-4a93-9e61-be5cb06dc249)

Also I improved like that:

```suggestion
    abort_with0("Skipping this step as the current Java version is already the selected version: #{ac_selected_java_version}.")
```

Reply:
The last change is more clear actually. Thnx :)